ci/requirements/upstream-dev.yml

```diff
@@ -0,0 +1,49 @@
+name: xarray-tests
+channels:
+  - conda-forge
+dependencies:
+  - boto3
+  - bottleneck
+  - cartopy
+  - cdms2
+  - cfgrib
+  - cftime
+  - coveralls
+  - dask
+  - distributed
+  - h5netcdf
+  - h5py=2
+  - hdf5
+  - hypothesis
+  - iris
+  - lxml    # Optional dep of pydap
+  - matplotlib
+  - nc-time-axis
+  - netcdf4
+  - numba
+  - numpy
+  - pandas
+  - pint
+  - pip
+  - pseudonetcdf
+  - pydap
+  # - pynio: not compatible with netCDF4>1.5.3; only tested in py36-bare-minimum
+  - pip
+  - pytest
+  - pytest-cov
+  - pytest-env
+  - rasterio
+  - scipy
+  - seaborn
+  - setuptools
+  - sparse
+  - toolz
+  - pip:
+    - numbagg
+    - git+https://github.com/dask/dask 
+    - git+https://github.com/dask/distributed 
+    - git+https://github.com/zarr-developers/zarr 
+    - git+https://github.com/Unidata/cftime 
+    - git+https://github.com/mapbox/rasterio 
+    - git+https://github.com/hgrecco/pint 
+    - git+https://github.com/pydata/bottleneck
```

Comment:
@keewis, let me know if these lines need to be replaced with

```bash
 python -m pip install \
        --no-deps \
        --upgrade \
        git+https://github.com/dask/dask \
        git+https://github.com/dask/distributed \
        git+https://github.com/zarr-developers/zarr \
        git+https://github.com/Unidata/cftime \
        git+https://github.com/mapbox/rasterio \
        git+https://github.com/hgrecco/pint \
        git+https://github.com/pydata/bottleneck
```
to match what is done in the existing upstream dev CI.

Reply:
Not sure. The current setup installs everything and then force-removes the ones we reinstall. That way, binary dependencies are installed using `conda` and only the libraries themselves are installed using `pip`, which might be advantageous for dependencies that don't have wheels.

If we decide to reproduce the current setup it would be nice to put the commands in a separate script and call that from both setups. In addition to keeping both CI in sync this would also help when trying to reproduce failures locally: right now setting up the environment involves copy/pasting commands.

Reply:
Addressed in [`dc6ba90` (#4583)](https://github.com/pydata/xarray/pull/4583/commits/dc6ba9083bab51f69e0da674f2db3f5b7a18bc01)